lib/mocktail/imitates_type/makes_double/declares_dry_class.rb

```diff
@@ -42,21 +44,28 @@ def define_double_methods!(dry_class, type, instance_methods)
       handles_dry_call = @handles_dry_call
       instance_methods.each do |method|
         dry_class.undef_method(method) if dry_class.method_defined?(method)
+        parameters = type.instance_method(method).parameters
+        signature = @transforms_params.transform(Call.new, params: parameters)
+        method_signature = @builds_method_signature.build(signature)
 
-        dry_class.define_method method, ->(*args, **kwargs, &block) {
-          Debug.guard_against_mocktail_accidentally_calling_mocks_if_debugging!
-          handles_dry_call.handle(Call.new(
-            singleton: false,
-            double: self,
-            original_type: type,
-            dry_type: dry_class,
-            method: method,
-            original_method: type.instance_method(method),
-            args: args,
-            kwargs: kwargs,
-            block: block
-          ))
-        }
+        # TODO: This is failing because none of self, type, dry_class, method,
+        # etc are in scope for the dry_class.
+        dry_class.class_eval <<-RUBY, __FILE__, __LINE__ + 1
+          def #{method}#{method_signature}
```

Comment:
IIRC the point of switching to class_eval was specifically to be able to use the method_signature _String_ which correctly represents the signature of the original method, rather than using `(*args, **kwargs, &block)` which is what's causing us to treat trailing hash args as kwargs (or was it the other way around?)

Reply:
Maybe? I honestly don't recall off-hand but can dive into this or anything else with you if it would be helpful